Stripe/PublicHeaders/STPAuthenticationContext.h

```diff
@@ -24,6 +25,17 @@ NS_ASSUME_NONNULL_BEGIN
  */
 - (UIViewController *)authenticationPresentingViewController;
 
+@optional
+/**
+ This method is called before presenting a UIViewController for authentication.
+
+ Implement this method if your customer is using Apple Pay.  For security, it's impossible to present UIViewControllers above the Apple Pay sheet.
+ This method should dismiss the PKPaymentAuthorizationViewController and call `continueBlock` in the dismissal's completion block.
+ 
+ Note that `paymentAuthorizationViewControllerDidFinish` is not called after `PKPaymentAuthorizationViewController` is dismissed.
+ */
+- (void)authenticationWillPresent:(STPVoidBlock)continueBlock;
```

Comment:
hmm, if we want to enforce that the merchant continues the flow by calling the block, `willPresent` seems semantically wrong...What do you think about
`- (void)prepareAuthenticationContextForPresentation:(STPVoidBlock)completion`
or similar?
I think `willPresent` has a connotation that no matter what I do this is happening, but in this case nothing will happen until you call the block

Reply:
Yeah, I think that works.  Definitely want to avoid that failure mode where the user doesn't call the block!